src/components/schedule/channelList.js

```diff
@@ -1,17 +1,28 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import ChannelListItem from './channelListItem';
+import { FETCH_STATES } from '../constants';
 
-const ChannelList = ({ channels }) => (
+const renderList = channels => channels.map(channel =>
+  <ChannelListItem key={channel.id} {...channel} />
+);
+
+const ChannelList = ({ channels, fetchState }) => (
   <div>
-    { channels.map(channel =>
-      <ChannelListItem key={channel.id} {...channel} />
-    ) }
+    {(fetchState === FETCH_STATES.FAILED) ?
```

Comment:
rzeczywiście lepiej to wygląda tak jak mówisz, przejrze pod tym kątem inne fragmenty kodu